mongodb/vibe/db/mongo/collection.d

```diff
@@ -259,6 +259,17 @@ struct MongoCollection {
 
 		Standards: $(LINK https://www.mongodb.com/docs/manual/reference/command/update/)
 	*/
+	UpdateResult replaceOne(T, U)(T filter, U replacement, ReplaceOptions options)
+	@safe {
+		UpdateOptions uoptions;
+		static foreach (i, val; options.tupleof)
+			uoptions.tupleof[i] = val;
```

Comment:
This looks very risky in terms of future changes (and off by one?). I'd suggest a by-name copy:

```D
static foreach (f; FieldNameTuple!ReplaceOptions)
	__traits(getMember, uoptions, f) = __traits(getMember, options, f);
```

Reply:
ah you are right sorry - I ended up not using this actually since I was then just using UpdateOptions, but I'll just add a sample unittest here to make sure it compiles